### DISCUSSION TITLE
Implementation of adding background noise

### DISCUSSION THREAD

**Discussion by matln** [speechbrain/speechbrain#1630] [Q&A]
Hi, I am confused about the implementation of `AddNoise` in `speechbrain/processing/speech_augmentation.py`:
1. The amplitude is obtained by computing the L1 Norm of the waveform, i.e., `out = torch.mean(torch.abs(waveforms), dim=1, keepdim=True)`. However, in [Signal-to-noise ratio wiki](https://en.wikipedia.org/wiki/Signal-to-noise_ratio), it mentions that the amplitude is computed by root mean square (RMS). Is this a bug? Or I misunderstood.
2. The clean and noise signals are scaled to maintain the amplitude after adding noise. But in other repos (e.g. [audiomentations](https://github.com/iver56/audiomentations/blob/b8a671d6d5ba4c3b3a5b618e7c0084c30aeec59b/audiomentations/augmentations/add_background_noise.py#L161), [AugLy](https://github.com/facebookresearch/AugLy/blob/c7ace6df316e79b44d5caae4e36521ea8e0d19e1/augly/audio/functional.py#L108)), the clean and noise signals are added directly. Does scaling improve data augmentation?
      ```python
      # Scale clean signal appropriately
      noisy_waveform *= 1 - noise_amplitude_factor
      
      # Rescale and add
      noise_amplitude = compute_amplitude(noise_waveform, noise_length)
      noise_waveform *= new_noise_amplitude / (noise_amplitude + 1e-14)
      noisy_waveform += noise_waveform
      ```
